.github/workflows/boards_build.yml

```diff
@@ -0,0 +1,36 @@
+---
+name: Build supported boards
+on:
+  push:
+    paths:
+      - 'patches/tock/**'
+      - 'third_party/**'
```

Comment:
@jmichelp will this pick up changes in git submodules?

Reply:
Also, should we add `deploy.py` or more generally any `.sh` or `.py` script, because they may affect the list of supported boards?

Reply:
Building a kernel for a given board is unrelated to `deploy.py`. The only script that is required is `setup.sh` because it applied kernel patches.
But we can add a flag to `deploy.py` to output the supported boards and iterate on them instead.

It also doesn't depend on `libtock-rs` to we should modify the pattern to be `third_party/tock/*` instead

Reply:
> It also doesn't depend on `libtock-rs` to we should modify the pattern to be `third_party/tock/*` instead

Why a single star in that case? You mentioned:

> We don't have subdirectory in patches/ so you can remove a star in the pattern here.

but isn't there sub-directories in third_party/tock? Or is the submodule treated as a single file?